sdk/samples/iot/hub/src/paho_iot_hub_pnp_example.c

```diff
@@ -0,0 +1,744 @@
+  AZ_RETURN_IF_FAILED(az_json_parser_parse_token(&jp, &tm.token));
+  if (tm.token.kind != AZ_JSON_TOKEN_BEGIN_OBJECT)
+  {
+    return AZ_ERROR_PARSER_UNEXPECTED_CHAR;
+  }
+
+  az_span start_time_span = AZ_SPAN_NULL;
+  while (az_succeeded(az_json_parser_parse_token_member(&jp, &tm)))
+  {
+    if (az_span_is_content_equal(report_method_payload_value_span, tm.name))
+    {
+      AZ_RETURN_IF_FAILED(az_json_token_get_string(&tm.token, &start_time_span));
+      break;
+    }
+
+    // else ignore token.
+  }
+
+  // Get the current time as a string
+  time_t rawtime;
+  struct tm* timeinfo;
+  time(&rawtime);
+  timeinfo = localtime(&rawtime);
+  size_t len = strftime(end_time_buffer, sizeof(end_time_buffer), "%Y-%m-%dT%H:%M:%S%z", timeinfo);
+  az_span time_span = az_span_init((uint8_t*)end_time_buffer, (int32_t)len);
+
+  // Build the method response payload
+  az_json_builder json_builder;
+  AZ_RETURN_IF_FAILED(az_json_builder_init(&json_builder, response, NULL));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_begin_object(&json_builder));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(&json_builder, report_max_temp_name_span));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_int32_number(&json_builder, (int32_t)device_max_temp));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(&json_builder, report_min_temp_name_span));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_int32_number(&json_builder, (int32_t)device_min_temp));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(&json_builder, report_avg_temp_name_span));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_int32_number(&json_builder, (int32_t)device_avg_temp));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(&json_builder, report_start_time_name_span));
+
+  // If the user specified a time, use that as the start | Otherwise use boot time
+  if (az_span_size(start_time_span) > 0)
+  {
+    AZ_RETURN_IF_FAILED(az_json_builder_append_string(&json_builder, start_time_span));
+  }
+  else
+  {
+    AZ_RETURN_IF_FAILED(az_json_builder_append_string(&json_builder, boot_time_span));
+  }
+
+  AZ_RETURN_IF_FAILED(az_json_builder_append_property_name(&json_builder, report_end_time_name_span));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_string(&json_builder, time_span));
+  AZ_RETURN_IF_FAILED(az_json_builder_append_end_object(&json_builder));
+
+  *out_response = az_json_builder_get_json(&json_builder);
+
+  return AZ_OK;
+}
+
+static int send_method_response(
+    az_iot_hub_client_method_request* request,
+    uint16_t status,
+    az_span response)
+{
+  // Get the response topic to publish the method response
+  if (az_failed(az_iot_hub_client_methods_response_get_publish_topic(
+          &client,
+          request->request_id,
+          status,
+          methods_response_topic,
+          sizeof(methods_response_topic),
+          NULL)))
+  {
+    printf("Unable to get twin document publish topic");
+    return -1;
+  }
```

Comment:
I would get the return code of the az_iot_hub_client_methods_response_get_publish_topic() and return that instead of -1.